src/compiler/utilities.ts

```diff
@@ -1504,7 +1529,7 @@ namespace ts {
     }
 
     // True if the given identifier, string literal, or number literal is the name of a declaration node
-    export function isDeclarationName(name: Node): name is Identifier | StringLiteral | LiteralExpression {
+    export function isDeclarationName(name: Node): boolean {

```

Comment:
Yes, @ivogabe is exactly right about why I removed the type predicate annotation.